research/ag_news/dynamic_layer_exchange/server.py

```diff
@@ -64,87 +58,38 @@ def fit_config(
         batch_size,
         num_classes,
-        sequence_length,
         normalize,
         filter_by_percentage,
-        norm_threshold,
-        exchange_percentage,
+        select_drift_more,
         sample_percentage,
         beta,
     )
 
 
-def eval_config(
-    local_steps: int,
-    batch_size: int,
-    num_classes: int,
-    sequence_length: int,
-    normalize: bool,
-    filter_by_percentage: bool,
-    norm_threshold: float,
-    exchange_percentage: float,
-    sample_percentage: float,
-    beta: float,
-    n_server_rounds: int,
-    current_round: int,
-) -> Config:
-    config = construct_config(
-        current_round,
-        local_steps,
-        batch_size,
-        num_classes,
-        sequence_length,
-        normalize,
-        filter_by_percentage,
-        norm_threshold,
-        exchange_percentage,
-        sample_percentage,
-        beta,
-    )
-    config["testing"] = n_server_rounds == current_round
-    return config
-
-
 def main(config: Dict[str, Any], server_address: str) -> None:
     # This function will be used to produce a config that is sent to each client to initialize their own environment
     fit_config_fn = partial(
         fit_config,
         config["local_steps"],
         config["batch_size"],
         config["num_classes"],
-        config["sequence_length"],
         config["normalize"],
         config["filter_by_percentage"],
-        config["norm_threshold"],
-        config["exchange_percentage"],
+        config["select_drift_more"],
         config["sample_percentage"],
         config["beta"],
     )
 
-    eval_config_fn = partial(
-        eval_config,
-        config["local_steps"],
-        config["batch_size"],
-        config["num_classes"],
-        config["sequence_length"],
-        config["normalize"],
-        config["filter_by_percentage"],
-        config["norm_threshold"],
-        config["exchange_percentage"],
-        config["sample_percentage"],
-        config["beta"],
-        config["n_server_rounds"],
+    initial_model = BertForSequenceClassification.from_pretrained(
+        "google-bert/bert-base-cased", num_labels=config["num_classes"]
     )
 
-    classifier_head = RobertaClassificationHead(num_classes=config["num_classes"], input_dim=768)
-    initial_model = ROBERTA_BASE_ENCODER.get_model(head=classifier_head)
-
     # Since clients can send different tensors to the server, we perform weighted averaging separately on each tensor.
     strategy = FedAvgDynamicLayer(
         # Server waits for min_available_clients before starting FL rounds
         min_available_clients=config["n_clients"],
         on_fit_config_fn=fit_config_fn,
-        on_evaluate_config_fn=eval_config_fn,
+        on_evaluate_config_fn=fit_config_fn,
         fit_metrics_aggregation_fn=fit_metrics_aggregation_fn,
         evaluate_metrics_aggregation_fn=evaluate_metrics_aggregation_fn,
         initial_parameters=get_all_model_parameters(initial_model),
```

Comment:
Not a big deal, but similar to the sparse tensor exchanger, I think you can just use a `SimpleClientManager` here instead of a `PoissonSamplingClientManager`